lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2440,68 +2424,134 @@ constructTransaction ctx genChange knownPools getPoolStatus (ApiT wid) body = do
             else
                 pure (txCtx, Nothing)
 
-        let runSelection outs =
-                W.selectAssets @_ @_ @s @k @ktype
-                  wrk era pp selectAssetsParams transform
-              where
-                selectAssetsParams = W.SelectAssetsParams
-                    { outputs = outs
-                    , pendingTxs
-                    , randomSeed = Nothing
-                    , txContext = txCtx'
-                    , utxoAvailableForInputs =
-                        UTxOSelection.fromIndex utxoAvailable
-                    , utxoAvailableForCollateral =
-                        UTxOIndex.toMap utxoAvailable
-                    , wallet
-                    , selectionStrategy = SelectionStrategyOptimal
+        outs <- case body ^. #payments of
+            Nothing -> pure []
+            Just (ApiPaymentAddresses content) ->
+                pure $ F.toList (addressAmountToTxOut <$> content)
+
+        let mintWithAddress
+                (ApiMintBurnData _ _ (ApiMint (ApiMintData (Just _) _)))
+                = True
+            mintWithAddress _ = False
+        let mintingOuts = case mintingBurning' of
+                Just mintBurns ->
+                    coalesceTokensPerAddr $
+                    map (toMintTxOut (fromJust policyXPubM)) $
+                    filter mintWithAddress $
+                    NE.toList mintBurns
+                Nothing -> []
+
+        unbalancedTx <- liftHandler $
+            W.constructTransaction @n @'CredFromKeyK
+                txLayer netLayer db wid era txCtx' PreSelection
+                    { outputs = outs <> mintingOuts
+                    , assetsToMint = fst $ txCtx' ^. #txAssetsToMint
+                    , assetsToBurn = fst $ txCtx' ^. #txAssetsToBurn
+                    , extraCoinSource = fromMaybe (Coin 0) refund
+                    , extraCoinSink = fromMaybe (Coin 0) deposit
```

Comment:
🤔 All these aside from `outputs` could be `mempty` (mkUnsignedTx doesn't look at them & ran integration tests - it works), so it seems the `PreSelection` datatype could be replaced with just `[TxOut]`. (Or possibly integrated with `TransactionCtx` 🤷‍♂️ )

@Unisay I recall you asked about this and I might have mislead you: it's _these_ similarly named fields https://github.com/input-output-hk/cardano-wallet/blob/7e7a60b9a3f23b7b726157fb721f1c1dfb7805a2/lib/wallet/src/Cardano/Wallet/CoinSelection.hs#L280-L290 in `SelectionParams` that currently are unavoidable, but the ones in `PreSelection` seem redundant.

Reply:
(Feel free to ignore for now and tackle when/how you see fit)